synapse/handlers/event_auth.py

```diff
@@ -93,13 +116,31 @@ async def check_restricted_join_rules(
             if not isinstance(space_id, str):
                 continue
 
-            # The user was joined to one of the spaces specified, they can join
-            # this room!
-            if space_id in joined_rooms:
-                return
+            result.append(space_id)
 
-        # The user was not in any of the required spaces.
-        raise AuthError(
-            403,
-            "You do not belong to any of the required spaces to join this room.",
-        )
+        return result
+
+    async def is_user_in_rooms(self, room_ids: Collection[str], user_id: str) -> bool:
```

Comment:
I couldn't find a method on any of the handlers / stores that lets you check if a user is in a room already, arguably this might be better on the `RoomMemberHandler` or something, but it is convenient for the `EventAuthHandler` to have no dependencies.